pkg/minikube/service/service.go

```diff
@@ -64,18 +66,19 @@ func (k *K8sClientGetter) GetCoreClient() (corev1.CoreV1Interface, error) {
 
 func (*K8sClientGetter) GetClientset() (*kubernetes.Clientset, error) {
 	loadingRules := clientcmd.NewDefaultClientConfigLoadingRules()
+	profile := viper.GetString(config.MachineProfile)
 	configOverrides := &clientcmd.ConfigOverrides{
 		Context: clientcmdapi.Context{
-			Cluster:  "minikube",
+			Cluster:  profile,
 			AuthInfo: "minikube",
```

Comment:
`AuthInfo` also uses the profile name, so `AuthInfo` should also be set to `profile` on line 70, see the function `PopulateKubeConfig` https://github.com/kubernetes/minikube/blob/master/pkg/util/kubeconfig/config.go#L77-L82.

The following is the result when `AuthInfo` is set to `minikube`:
```
$ out/minikube config set bootstrapper kubeadm
$ out/minikube profile cluster2
minikube profile was successfully set to cluster2
$ out/minikube start
Starting local Kubernetes v1.8.0 cluster...
Starting VM...
Getting VM IP address...
Moving files into cluster...
Downloading kubeadm v1.8.0
Downloading kubelet v1.8.0
Finished Downloading kubelet v1.8.0
Finished Downloading kubeadm v1.8.0
Setting up certs...
Connecting to cluster...
Setting up kubeconfig...
Starting cluster components...
E1120 08:51:44.524528    5936 start.go:265] Error starting cluster:  timed out waiting to unmark master: getting core client: getting clientset: Error creating kubeConfig: auth info "minikube" does not exist

```